src/blas/backends/cublas/cublas_scope_handle.hpp

```diff
@@ -84,22 +63,15 @@ the handle must be destroyed when the context goes out of scope. This will bind
 **/
 
 class CublasScopedContextHandler {
-    CUcontext original_;
-    sycl::context* placedContext_;
-    bool needToRecover_;
     sycl::interop_handle& ih;
-#ifdef ONEMKL_PI_INTERFACE_REMOVED
-    static thread_local cublas_handle<ur_context_handle_t> handle_helper;
```

Comment:
It seems that you are removing all uses of `cublas_handle`, meaning the class is never instantiated. Is this intended?

Reply:
I only did that in the last commit when I switched to using the `shared_ptr` instead of `cublas_handle`. After that change we don't need `cublas_handle` at all. But I'm not sure if we want to do that. 

The only problem with sticking to `cublas_handle` is that if I modify it's definition ([removing std::atomic commit](https://github.com/konradkusiak97/oneMKL/pull/1/commits/0eb8feaf0a794e82d079e2eb6e88ba9340808be2)) then we are breaking HIPSYCL implementation and would need to fix that.

And the base of all those changes is the fact that we only use cuda primary ctx in DPC++, I don't know what HIPSYCL does so maybe they need the `std::atomic` in the `cublas_handle`.